### PR TITLE
Opsgenie integration

### PR DESCRIPTION
**Pull request by fernandosure**
there was no integration with opsGenie so it would be nice if everyone could take advantage of this instead of making it from scratch

### CODE REVIEW

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/config/NotifierConfiguration.java [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
 import java.util.List;
 
+import de.codecentric.boot.admin.notify.*;
```
Comment (joshiste) [inline spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/config/NotifierConfiguration.java]: Please remove the wildcard import

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/config/NotifierConfiguration.java [hunk 2]
```diff
@@ -136,6 +129,23 @@ public PagerdutyNotifier pagerdutyNotifier() {
...
 
+
+	@Configuration
+	@ConditionalOnProperty(prefix = "spring.boot.admin.notify.opsgenie", name = "api-key")
+	@AutoConfigureBefore({ NotifierListenerConfiguration.class,
+			CompositeNotifierConfiguration.class })
+	public static class OpsGenieNotifierConfiguration {
+		@Bean
+		@ConditionalOnMissingBean
+		@ConditionalOnProperty(prefix = "spring.boot.admin.notify.opsgenie", name = "enabled", matchIfMissing = true)
```
Comment (joshiste) [inline spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/config/NotifierConfiguration.java]: All other notifier-beans are instantiated even if they are disabled please align to that.

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/notify/OpsGenieNotifier.java [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+     */
+    private Expression description;
+
+    public OpsGenieNotifier() {
+        this.description = parser.parseExpression(DEFAULT_MESSAGE, ParserContext.TEMPLATE_EXPRESSION);
+    }
+
+    @Override
+    protected void doNotify(ClientApplicationEvent event) throws Exception {
+        restTemplate.postForEntity(buildUrl(event), createEvent(event), Void.class);
```
Comment (joshiste) [inline spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/notify/OpsGenieNotifier.java]: Please add a correct content-type to the request. Depending on the order of the MessageConverters it happens that the requests end up being in xml. If you need a hint doing that please have a look at the LetsChatNotifier.

File: spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/config/NotifierConfigurationTest.java [hunk 4]
```diff
@@ -25,6 +25,7 @@
 import java.util.Arrays;
 import java.util.List;
 
+import de.codecentric.boot.admin.notify.*;
```
Comment (joshiste) [inline spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/config/NotifierConfigurationTest.java]: Please remove the wildcard import

**Comment by fernandosure** [comment]
Hey @joshiste can you please review my recent changes!

**Comment by joshiste** [comment]
merged with 230896f0a840371f65a1c2b2ed56dd9b16ed106e. Thanks for contributing!

**Pull request closed without merging**